packages/headless/src/controllers/search-parameter-manager/headless-search-parameter-manager.ts

```diff
@@ -143,3 +145,22 @@ function getSortCriteria(state: Partial<SearchParametersState>) {
   const shouldInclude = sortCriteria !== getSortCriteriaInitialState();
   return shouldInclude ? {sortCriteria} : {};
 }
+
+function getFacets(state: Partial<SearchParametersState>) {
+  if (state.facetSet === undefined) {
+    return {};
+  }
+
+  const f = Object.entries(state.facetSet)
+    .map(([facetId, request]) => {
+      const selectedValues = getSelectedValues(request.currentValues);
+      return selectedValues.length ? {[facetId]: selectedValues} : {};
+    })
+    .reduce((acc, obj) => ({...acc, ...obj}), {});
+
+  return Object.keys(f).length ? {f} : {};
+}
+
+function getSelectedValues(values: FacetValueRequest[]) {
```

Comment:
I think this could benefit to be made available in one of the different `facet-selector` file we have, no ?

It's seems close to what the breadcrumb does.

Reply:
It is similar, but uses the request to determine selected values rather than the response. The url will reflect the state of the page as soon as a facet value is selected/unselected, just like on Google, Amazon and the JSUI.

The `SearchParametersState` interface used in this file also includes only the request slices. The `response` property is under the `search` response slice, so it is not possible to use the existing selector with the `state` in this file.

Personally, I would wait until the same functionality is needed elsewhere to refactor, because it may not be/all other places are using the response.